File: src/domain/models/lead.py
import datetime
import uuid
from typing import Optional


class Lead:
    def __init__(
        self,
        org_id: uuid.UUID,
        session_id: uuid.UUID,
        name: Optional[str] = None,
        phone: Optional[str] = None,
        email: Optional[str] = None,
        intent: Optional[str] = None,
        product_interest: Optional[str] = None,
        budget: Optional[float] = None,
        id: Optional[uuid.UUID] = None,
        created_at: Optional[datetime.datetime] = None,
    ):
        self.id = id or uuid.uuid4()
        self.org_id = org_id
        self.session_id = session_id
        self.name = name
        self.phone = phone
        self.email = email
        self.intent = intent
        self.product_interest = product_interest
        self.budget = budget
        self.created_at = created_at or datetime.datetime.utcnow()

File: control_center/lib/features/leads/domain/models/lead.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'lead.freezed.dart';
part 'lead.g.dart';

enum LeadStatus {
  newLead,
  contacted,
  qualified,
  proposalSent,
  negotiation,
  won,
  lost,
}

enum LeadSource { whatsapp, callingAgent, website, manual, import }

@freezed
class Lead with _$Lead {
  const factory Lead({
    required String id,
    required String orgId,
    required String name,
    String? email,
    String? phone,
    String? company,
    String? jobTitle,
    @Default(LeadStatus.newLead) LeadStatus status,
    @Default(LeadSource.manual) LeadSource source,
    String? assignedTo,
    String? assignedToName,
    @Default(0) int aiScore,
    @Default(0) int intentScore,
    @Default(0) int budgetScore,
    @Default(0) int engagementScore,
    String? notes,
    Map<String, dynamic>? metadata,
    String? conversationId,
    DateTime? lastContactedAt,
    DateTime? qualifiedAt,
    DateTime? wonAt,
    DateTime? lostAt,
    required DateTime createdAt,
    required DateTime updatedAt,
  }) = _Lead;

  factory Lead.fromJson(Map<String, dynamic> json) => _$LeadFromJson(json);
}
